### ISSUE TITLE
Question regarding 'requires_grad' on TensorBase

### ISSUE THREAD

**Issue by eseaflower** [fastai/fastai2#388]
Hi!
I encounter an unexpected behaviour when using instances of TensorBase (or derived classes TensorPoint/TensorBBox) in computations. It seems that gradients are not propagated when using these classes. Here is a gist describing what I mean, https://gist.github.com/eseaflower/a675c534dcd933c4315653c1d5c4b028

I expected that both z1 and z2 in the example should have the grad_fn set. This poses a problem when using for instance TensorBBox as targets in a Learner. It can be noted that if I use 'native' torch-functions (F.mse_loss for instance) the gradient is set as expected.

I don't know if the observed behaviour is by design or not, if it is I think it should be mentioned in the docs somewhere that TensorBase-instances need to be casted to torch.Tensor before doing computations with them.

**Comment by cwza** [comment]
Exactly the same problem I asked before. https://github.com/fastai/fastai2/issues/278
We can't compute gradient w.r.t. Tensor subclass due to the restriction of pytorch.
This behavior is really confused so I have the same opinion as @eseaflower that it is worth to be mentioned in docs somewhere.

My workaround now is to write a callback that cast xb or yb to plain tensor before each batch.

**Comment by eseaflower** [comment]
Thanks @cwza for pointing to the existing issue. I'll probably use a callback to cast my values before applying the model.

**Comment by muellerzr** [comment]
A PR on documentation is always welcome :smile: I'd recommend a quick note in vision.core and text.data where the various `TensorX` classes are defined 😄

**Comment by jph00** [comment]
That should be fixed now. Many thanks for the report.
